src/script/lua_api/l_mapgen.cpp

```diff
@@ -450,6 +451,32 @@ size_t get_biome_list(lua_State *L, int index,
 
 ///////////////////////////////////////////////////////////////////////////////
 
+// get_biome_id(biomename)
+// returns the biome id used in biomemap
+int ModApiMapgen::l_get_biome_id(lua_State *L)
+{
+	//std::string val("new biome function test");
+	const char *biome_str = lua_tostring(L, 1);
+	if (!biome_str) {
+		lua_pushnil(L);
+		return 1;
+	}
+
+	BiomeManager *bmgr    = getServer(L)->getEmergeManager()->biomemgr;
+
+	if (!bmgr)
+		return 0;
+
+	Biome *biome = (Biome *) bmgr->getByName(biome_str);
+	if (biome && biome->index)
```

Comment:
I'm not sure, but from shallow reading of the code, you should compare the index with `OBJDEF_INVALID_INDEX`, if you want to check whether the index is valid.